oxide-openapi-gen-ts/src/client/api.ts

```diff
@@ -76,6 +77,45 @@ export function copyStaticFiles(destDir: string) {
   copyFile("./static/http-client.ts", destDir);
 }
 
+export function genPathParams(
+  params: Param[],
+  opName: string,
+  schemaNames: string[],
+  io: IO
+) {
+  io.w(`export interface ${pathParamsType(opName)} {`);
+  for (const param of params) {
+    if ("description" in param.schema || "title" in param.schema) {
+      docComment(extractDoc(param.schema), schemaNames, io);
+    }
+    io.w0(`  ${processParamName(param.name)}:`);
+    schemaToTypes(param.schema, io);
+    io.w(",");
+  }
+  io.w("}\n");
+}
+
+export function genQueryParams(
+  params: Param[],
+  opName: string,
+  schemaNames: string[],
+  io: IO
+) {
+  io.w(`export interface ${queryParamsType(opName)} {`);
+  for (const param of params) {
+    if ("description" in param.schema || "title" in param.schema) {
+      docComment(extractDoc(param.schema), schemaNames, io);
+    }
+
+    io.w0(`  ${processParamName(param.name)}`);
+    if (!param.required) io.w0("?");
+    io.w0(": ");
+    schemaToTypes(param.schema, io);
+    io.w(",");
+  }
+  io.w("}\n");
+}
+
```

Comment:
I extracted these in anticipation of unit testing them as well, but I'm not doing that yet. I don't think I'm actually going to do anything with `nullable` for path and query param types because you can't actually pass `null` to them (it is not distinguishable from the string `"null"`). We might want to get those `nullable` out of the API spec if they don't mean anything.